main.go

```diff
@@ -301,14 +301,20 @@ func init() {
 		}
 	}
 
-	if config.GCP.PubSub.ProjectID != "" && config.GCP.PubSub.Topic != "" && config.GCP.Credentials != "" {
+	if (config.GCP.PubSub.ProjectID != "" && config.GCP.PubSub.Topic != "") || config.GCP.Storage.Bucket != "" {
 		var err error
 		gcpClient, err = outputs.NewGCPClient(config, stats, promStats, statsdClient, dogstatsdClient)
 		if err != nil {
 			config.GCP.PubSub.ProjectID = ""
 			config.GCP.PubSub.Topic = ""
+			config.GCP.Storage.Bucket = ""
 		} else {
-			outputs.EnabledOutputs = append(outputs.EnabledOutputs, "GCPPubSub")
+			if config.GCP.PubSub.Topic != "" && config.GCP.PubSub.ProjectID != "" {
+				outputs.EnabledOutputs = append(outputs.EnabledOutputs, "GCPPubSub")
+			}
+			if config.GCP.Storage.Bucket != "" {
+				outputs.EnabledOutputs = append(outputs.EnabledOutputs, "GCPStorage")
+			}
```

Comment:
I think we have a bad pattern here. My point is, if we have `config.GCP.PubSub.ProjectID != "" && config.GCP.PubSub.Topic != "") || config.GCP.Storage.Bucket != ""`, then `outputs.EnabledOutputs = append(outputs.EnabledOutputs,` will be called twice. One for `PubSub` and one for `Storage`. You see what I mean?

Reply:
Nevermind, I commented too quickly.